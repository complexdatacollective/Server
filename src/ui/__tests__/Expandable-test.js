/* eslint-env jest */

import React from 'react';
import { shallow } from 'enzyme';
import Expandable from '../Expandable';

describe('<Expandable />', () => {
  it('should render', () => {
    const subject = shallow((
      <Expandable className="foo">
        Bar
      </Expandable>
    ));

    expect(subject).toMatchSnapshot();
  });

  it('should have expandable classes', () => {
    const subject = shallow((
      <Expandable>
        Bar
      </Expandable>
    ));

    expect(subject.prop('className')).toEqual('expandable');

    subject.setProps({ className: 'foo' });

    expect(subject.prop('className')).toEqual('foo expandable');

    subject.setProps({ className: 'foo', open: true });

    expect(subject.prop('className')).toEqual('foo expandable expandable--open');
  });
});
