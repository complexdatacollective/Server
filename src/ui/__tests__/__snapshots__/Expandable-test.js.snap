exports[`<Expandable /> should render 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div
    className="foo expandable">
    Bar
</div>,
  "nodes": Array [
    <div
      className="foo expandable">
      Bar
</div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 2,
      "_context": Object {},
      "_currentElement": <Expandable
        className="foo"
        open={false}>
        Bar
</Expandable>,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": StatelessComponent {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "children": "Bar",
          "className": "foo",
          "open": false,
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div
          className="foo expandable">
          Bar
</div>,
        "_debugID": 2,
        "_renderedOutput": <div
          className="foo expandable">
          Bar
</div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Expandable
    className="foo"
    open={false}>
    Bar
</Expandable>,
}
`;
